Sources/iPhone/Data Manager/Proxies/FilesProxy.m

```diff
@@ -368,6 +372,87 @@ -(NSString *)fileAction:(NSString *)protocol source:(NSString *)source destinati
 	return nil;
 }
 
+
+-(void) uploadFile:(NSData *) fileData asFileName:(NSString *) fileAttachName inFolder:(NSString *) currentFolder ofDrive:(NSString *) driveName {
+    /*
+     ECMS web service
+     1. Upload file POST
+     Query params: uploadId= : An arbitrary value to keep until the end &  action=upload
+     Content type: multipart/form-data; boundary= with an arbitrary boundary.
+     Body:
+     --BOUNDARY
+     Content-Disposition: form-data; name="file"; filename="..."  /!\ name must be "file"
+     Content-Type: the content type of the file to upload
+     
+     // File content
+     --BOUNDARY
+     2. save file GET /portal/rest/managedocument/uploadFile/control
+     uploadId= : the value chosen at step 1
+     action=save
+     workspaceName= : the workspace in which to move the file
+     driveName= : the drive, within the workspace, in which to move the file
+     currentFolder= : the folder, within the drive, in which to move the file
+     fileName= : the name of the file (can be different than the original)
+     */
+    NSString * serverURL = [ApplicationPreferencesManager sharedInstance].selectedAccount.serverUrl;
+    
+    NSString * uploadId = [NSUUID UUID].UUIDString;
+    uploadId = [uploadId stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
+    NSString * boundary = [NSString stringWithFormat:@"-----%@",uploadId];
+    
+    NSString * postRESTURL = [NSString stringWithFormat:@"%@/portal/rest/managedocument/uploadFile/upload?uploadId=%@&action=upload", serverURL, uploadId];
```

Comment:
URL endpoint should be defined in defines.h with other `DOCUMENT_...` constants, e.g.
`DOCUMENT_UPLOAD_SERVICE_PATH @"/portal/rest/managedocument/uploadFile/upload"`
and
`DOCUMENT_SAVE_SERVICE_PATH @"/portal/rest/managedocument/uploadFile/control"`